src/main/java/com/google/cloud/genomics/dataflow/pipelines/LoadReadsToBigQuery.java

```diff
@@ -0,0 +1,372 @@
+    }
+  }
+
+  public static void main(String[] args) throws GeneralSecurityException, IOException,
+         URISyntaxException {
+    // Register the options so that they show up via --help.
+    PipelineOptionsFactory.register(Options.class);
+    pipelineOptions = PipelineOptionsFactory.fromArgs(args)
+        .withValidation().as(Options.class);
+    // Option validation is not yet automatic, we make an explicit call here.
+    Options.Methods.validateOptions(pipelineOptions);
+    // Needed for BigQuery.
+    pipelineOptions.setTempLocation(pipelineOptions.getTempPath());
+
+    auth = GenomicsOptions.Methods.getGenomicsAuth(pipelineOptions);
+    p = Pipeline.create(pipelineOptions);
+
+    // Ensure data is accessible.
+    String BAMFilePath = pipelineOptions.getBAMFilePath();
+    if (!Strings.isNullOrEmpty(BAMFilePath)) {
+      if (!GCSURLExists(BAMFilePath)) {
+        System.out.println("Error: BAM file " + BAMFilePath + " not found. A BAM file is "
+            + "required.");
+        return;
+      }
+      if (pipelineOptions.getMaxShardSizeMB() > 0) {
+        // The BAM code expects an index at BAMFilePath+".bai"
+        // and sharded reading will fail if the index isn't there.
+        String BAMIndexPath = BAMFilePath + ".bai";
+        if (!GCSURLExists(BAMIndexPath)) {
+          System.out.println("Error: Index file " + BAMIndexPath + " not found. An index is required "
+            + "for sharded export.");
+          return;
+        }
+      }
+    }
+
+    PCollection<Read> reads = getReads();
+    // TODO(#221): validate the given BigQuery table, and fail fast if it is not valid and empty.
+    reads.apply(ParDo.of(new ReadToRowConverter()))
+        .apply(BigQueryIO.writeTableRows()
+            .to(pipelineOptions.getBigQueryTablePath())
+            .withCreateDisposition(BigQueryIO.Write.CreateDisposition.CREATE_IF_NEEDED)
+            .withSchema(ReadToRowConverter.getSchema()));
+
+    PipelineResult result = p.run();
+    if(pipelineOptions.getWait()) {
+      result.waitUntilFinish();
+    }
+  }
+
+  private static boolean GCSURLExists(String url) {
+    // Ensure data is accessible.
+    try {
+      // If we can read the size, then surely we can read the file.
+      GcsPath fn = GcsPath.fromUri(url);
+      Storage.Objects storageClient = GCSOptions.Methods.createStorageClient(pipelineOptions, auth);
+      Storage.Objects.Get getter = storageClient.get(fn.getBucket(), fn.getObject());
+      StorageObject object = getter.execute();
+      BigInteger size = object.getSize();
+      return true;
+    } catch (Exception x) {
+      System.out.print("Exception: ");
+      System.out.println(x.getMessage());
+      return false;
+    }
+  }
+
+  private static PCollection<Read> getReads() throws IOException, URISyntaxException {
+    if (!pipelineOptions.getBAMFilePath().isEmpty()) {
+      return getReadsFromBAMFile();
+    }
+    throw new IOException("BAM file must be specified");
```

Comment:
I'd think this is more of an IllegalArgumentException or something? Can it not be validated earlier?